yb-voyager/src/utils/recordReader.go

```diff
@@ -0,0 +1,39 @@
+package utils
+
+import (
+	"bufio"
+	"io"
+	"os"
+)
+
+type RecordReader struct {
+	file       *os.File
+	NumRecords int64
+	reader     *bufio.Reader
+}
+
+func (rr *RecordReader) Read(p []byte) (n int, err error) {
+	n, err = rr.reader.Read(p)
+
+	for i := 0; i < n; i++ {
```

Comment:
If the CSV file is UTF-8 encoded (which is essential), then this code may not be correct. Refer https://golangbyexample.com/iterate-over-a-string-golang/ . Reading each byte separately may lead to incorrectly classifying some bytes as `\n`.

Unfortunately, a for-range loop may not also be enough. Consider a case where only the first byte of a multi-byte character was included in the buffer `p`. In that case, the for loop will fail with "invalid byte in utf-8" error.

Given how low the probability of that happening is, for now, just add a comment stating this problem.

Reply:
Added